### PR TITLE
Pannable Map

### PR DESCRIPTION
**Pull request by Torpedo99**
# Description
Allows the game map to be pannable using the mouse/touch.

# Testing
### Using Mouse
The map can be panned with the mouse simply by holding left click and dragging the mouse around.
### Android
Map can be panned around by touching the screen and moving your finger around.

# Notes
Map panning should work fine on Android since I've used LibGDX's in-built [touchDragged](https://libgdx.badlogicgames.com/ci/nightlies/docs/api/com/badlogic/gdx/InputProcessor.html#touchDragged-int-int-int-) function. Although further testing would be appreciated since this is the first ever implementation of such.

# Outstanding work
- [x] Make sure grid map is drawn to fill the new panned screen.
- [x] Add compatibility with the new waypoint system.
- [x] Add a settings option for scroll speed.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by casals** [comment]
Hey @Torpedo99 - the scrollable map came out really nice. I just felt that using the mouse scrolls the map in very small increments - don't know if it's something related to how it behaves on Android (didn't test it) or if it's just here. Leaving the comment here anyway so others can follow. Perhaps including a configuration option for this would solve the problem (in the case it's platform-specific).

**Comment by Torpedo99** [comment]
> I just felt that using the mouse scrolls the map in very small increments

There does exist a  `SCROLL_SPEED` constant which I've calibrated myself. Although I assumed in case users wanted to scroll faster, they could zoom out to do so, and similarly, if they zoomed in, they probably would also want the map to scroll slower for accuracy. 
However I think that a settings option for the scroll speed is a very nice idea. I'll add it the to-do list for the PR :) 
@casals

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/GameOptions.java [hunk 1]
```diff
@@ -250,6 +255,13 @@ public void advanceMusicVolMul() {
         save();
     }
 
+    public void advanceMapScrollSpeed() {
+        mapScrollSpeed++;
+        if(mapScrollSpeed > 10) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/GameOptions.java]:
```suggestion
        if (mapScrollSpeed > 10) {
```

File: engine/src/main/java/org/destinationsol/game/screens/MapScreen.java [hunk 2]
```diff
@@ -102,10 +113,18 @@ public void updateCustom(SolApplication solApplication, SolInputManager.InputPoi
             }
         }
 
+        if(im.touchDragged) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/screens/MapScreen.java]:
```suggestion
        if (im.touchDragged) {
```

File: engine/src/main/java/org/destinationsol/menu/OptionsScreen.java [hunk 3]
```diff
@@ -113,6 +118,11 @@ public void updateCustom(SolApplication solApplication, SolInputManager.InputPoi
             options.advanceMusicVolMul();
             solApplication.getMusicManager().changeVolume(options);
         }
+
+        mapScrollSpeedControl.setDisplayName("Map Pan Speed: " + options.getMapScrollSpeed());
+        if(mapScrollSpeedControl.isJustOff()) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/menu/OptionsScreen.java]:
```suggestion
        if (mapScrollSpeedControl.isJustOff()) {
```

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**